src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8386,46 +8364,47 @@ public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
                 // It is possible for the awaiter type returned from GetAwaiter to not be a named type. e.g. it could be a type parameter.
                 // Proper handling of this is additional work which only benefits a very uncommon scenario,
                 // so we will just use the originally bound GetResult method in this case.
+                var getResult = awaitableInfo.GetResult!;
```

Comment:
suppression probably can be avoided if `node.AwaitableInfo.GetResult` is changed to `awaitableInfo.GetResult` in the `if`.